### DISCUSSION TITLE
Calling cargo make from task

### DISCUSSION THREAD

**Discussion by wmmc88** [sagiegurari/cargo-make#1072] [Q&A]
**TLDR**: Is there a way to call `cargo make` from a task that forwards everything (including env, cli args passed to cargo-make itself)? 

I have a situation where I have 2 separate workspaces with 2 separate (and different) Makefiles. The 2 workspaces share a common parent directory which has a 3rd makefile. In this 3rd makefile, I would like to have tasks to call the makefiles from the 2 workspaces (ie. subdirectories). I want to do it in a way where the init tasks of the sub-directories are called, the cargo-make setting passed to the parent makefile are forwarded (ex. `cargo make --no-workspace makefile-a-task` will forward `--no-workspace` to the makefile a task), and also want all cargo-make configurations forwarded (ex. i want `env` section in parent makefile to apply to the sub ones). This is what I have so far:

```toml
[tasks.sub-flow]
workspace = false
cwd = "./sub-dir"
# Use script instead of run_task to pass on `cwd` to the task
script = '''
#!@duckscript

exec --fail-on-error cargo make sub-dir-task-name
'''
```

This seems to forward `env`, and also run the init tasks for `sub-dir`, but it doesn't handle forwarding cargo-make cli args (i.e. `cargo make --env RUSTFLAGS="-D warnings" sub-flow` won't apply `RUSTFLAGS` to `sub-dir-task-name`). Is there a canonical way to handle this?

**Comment by sagiegurari** [comment]
i could be wrong but some env vars are not forwarded since they might be overwritten by the default env in stable.toml
could that be the issue?

**Reply by wmmc88** [reply]
Hmm. This does seem to be the case, since it works when forwarding stuff like `--env ABC="some val"`. Is there a way to prevent that overriding? or even better, a way for a task to get all of the arguments passed to cargo-make itself (i.e. --env, --no-workspace, etc) in some variable i can just pass to the script in the `sub-flow` task?
